services/ens/database.go

```diff
@@ -0,0 +1,57 @@
+package ens
+
+import (
+	"database/sql"
+)
+
+type Database struct {
+	db *sql.DB
+}
+
+type EnsUsernameDetails struct {
+	Username string `json:"username"`
+	ChainId  uint64 `json:"chainId"`
+}
+
+func NewEnsDatabase(db *sql.DB) *Database {
+	return &Database{db: db}
+}
+
+func (db *Database) GetEnsUsernames() (result []*EnsUsernameDetails, err error) {
+
+	const sqlQuery = `SELECT username, chain_id
+					  FROM ens_usernames`
+
+	rows, err := db.db.Query(sqlQuery)
+
+	if err != nil {
+		return result, err
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		var ensUsername EnsUsernameDetails
+		err = rows.Scan(&ensUsername.Username, &ensUsername.ChainId)
+		if err != nil {
+			return result, err
```

Comment:
```suggestion
			return nil, err
```
To avoid returning partial results